### PR TITLE
Remove check for clients in status command

### PR DESCRIPTION
**Pull request by t-eckert**
Changes proposed in this PR:
- Remove the check for Consul clients in the `consul-k8s status` command. Clients don't exist in agentless (RIP).

How I've tested this PR:
- Unit tests
- Manual install -> status check -> uninstall

How I expect reviewers to test this PR:
- Unit tests
- Manual install -> status check -> uninstall
- :eyes:

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: cli/cmd/status/status.go [hunk 1]
```diff
@@ -128,13 +126,6 @@ func (c *Command) Run(args []string) int {
 		c.UI.Output(s, terminal.WithSuccessStyle())
 	}
 
-	if s, err := c.checkConsulClients(namespace); err != nil {
-		c.UI.Output(err.Error(), terminal.WithErrorStyle())
-		return 1
-	} else {
-		c.UI.Output(s, terminal.WithSuccessStyle())
-	}
-
```
Comment (ishustava) [inline cli/cmd/status/status.go:137]: Should we instead check if clients are enabled and only check status then? It's also a bit weird that we check servers all the time even for clusters that might not have servers installed.

Comment (t-eckert) [reply cli/cmd/status/status.go:137]: Yeah, that's possible. We can have it read the deployment configuration and determine what to check then. That would be a separate set of work. Should I close this PR and take that approach instead?

Comment (ishustava) [reply cli/cmd/status/status.go:137]: Ok, sounds good. I'll approve this one - I think it's good to get at least that merged in.

**Review by ishustava — APPROVED** [review]

File: cli/cmd/status/status.go (hunk 1 above)
Comment (t-eckert) [reply cli/cmd/status/status.go:137]: https://hashicorp.atlassian.net/browse/NET-1037

**Review by wilkermichael — APPROVED** [review]

**Pull request merged**